java/arcs/core/data/proto/TypeProtoDecoders.kt

```diff
@@ -113,10 +113,14 @@ fun ReferenceTypeProto.decode() = ReferenceType(referredType.decode())
 fun TupleTypeProto.decode() = TupleType(elementsList.map { it.decode() })
 
 /** Converts a [TypeVariableProto] protobuf instance into a Kotlin [TypeVariable] instance. */
-fun TypeVariableProto.decode() = TypeVariable(
-    name,
-    if (hasConstraint()) constraint.constraintType.decode() else null
-)
+fun TypeVariableProto.decode(): TypeVariable {
+    require(hasConstraint()) { "TypeVariableProto must have a constraint." }
```

Comment:
Why is the constraint required?

Reply:
This just checks that there is a valid constraint field in the proto; not that the type variable always has a constraint.